### ISSUE TITLE
The mysql configuration properties that you recommend. one should not be recommended. 

### ISSUE THREAD

**Issue by miemieYaho** [brettwooldridge/HikariCP#1112]
#### Environment
```
HikariCP version: 2.7.8
JDK version     : 1.8.0_111
Database        : MySQL 5.7
Driver version  : 6.0.6
```
if i set the property of useLocalTransactionState value is true;
the spring transaction in some situation will not right;
my java project with spring(4.3.13) and mybatis(3.4.5),
spring transaction use the default,
if i insert tableA one or more data success,then insert tableB one date error,the data of tableA can't rollback;
but i insert tableA one or more data success,then insert tableB one date success and then insert tableB one data error,the data can rollback;

**Comment by billoneil** [comment]
This sounds more like a spring configuration or code issue not a HikariCP one. Is there anything that makes you think the issue is related to HikariCP specifically?

**Comment by billoneil** [comment]
Actually it looks to be a bug with the driver. https://bugs.mysql.com/bug.php?id=75209. It says it was fixed in 6.0.5 can you make sure nothing on the classpath is pulling in an older version.

**Comment by brettwooldridge** [comment]
@billoneil Thanks for the sleuthing and all of the help recently!

**Comment by miemieYaho** [comment]
@billoneil yes, it's not a bug with hikaricp, my mysql java driver version is 6.0.6,i just wan't tell you,you recommend properties of useLocalTransactionState have problem;
if set value is true , the transaction will not work right:
insert tableA one data success , then insert tableB one date success , but then insert tableB one data error ,rollback is not right ,the tableA have data, but tableB no data

**Comment by shyr** [comment]
Tha bug was fixed long time ago in Connector/J 5.1.40. 
Is `useLocalTransactionState=true` option recommendable again in the latest MySQL after the bug fixed version?
